mempool/reactor.go

```diff
@@ -88,7 +106,10 @@ func (memR *MempoolReactor) Receive(chID byte, src p2p.Peer, msgBytes []byte) {
 
 	switch msg := msg.(type) {
 	case *TxMessage:
-		err := memR.Mempool.CheckTx(msg.Tx, nil)
+		memR.idMtx.RLock()
+		peerID := memR.peerMap[src.ID()]
+		memR.idMtx.RUnlock()
```

Comment:
we have to use `defer` because CheckTx might panic and mempool will be frozen forever.

![attach_messages_flow_-_new_page](https://user-images.githubusercontent.com/1282182/48258265-f96dc680-e414-11e8-918e-c27fdcf31c34.png)

Reply:
Oh, you're right. I am becoming a bit paranoid :)